resources/assets/js/components/H2Components/Dialog.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from "react";
+import {
+  h2ComponentDialogLoad,
+  h2ComponentDialogLoadResize,
+  h2ComponentDialogEnableTrigger,
+} from "@hydrogen-design-system/system/dist/import/latest/components/dialog/scripts/dialog";
+import { defineMessages, useIntl } from "react-intl";
+import { GeneralProps, GeneralBtnProps } from "./utils";
+
+const messages = defineMessages({
+  closeDialog: {
+    id: "hydrogen.dialog.close",
+    defaultMessage: "Close",
+  },
+});
+
+interface DialogContext {
+  /** The dialogs id. */
+  id?: string;
+}
+
+const DialogContext = React.createContext<DialogContext | undefined>(undefined);
+
+/**
+ * This Context hook allows our child components to easily reach
+ * into the Tabs context and get the pieces it needs.
+ *
+ * Bonus: it even makes sure the component is used within a
+ * Dialog component!
+ */
+const useDialogContext = (): Partial<DialogContext> => {
+  const context = React.useContext(DialogContext);
+  if (!context) {
+    throw new Error("This component must be used within a <Dialog> component.");
+  }
+  return context;
+};
+
+const Actions: React.FunctionComponent<GeneralProps> = (props) => {
+  useDialogContext(); // Ensures sub-component can only be used within the Dialog component.
+  const { children } = props;
+  return (
+    <div data-h2-dialog-actions {...props}>
+      {children}
+    </div>
+  );
+};
+
+const ActionBtn: React.FunctionComponent<GeneralBtnProps> = (props) => {
+  const { id = "dialog" } = useDialogContext();
+  const { buttonStyling, type, onClick, children } = props;
+  const ref = React.useRef(null);
+  React.useEffect((): void => {
+    h2ComponentDialogEnableTrigger("latest", ref.current);
+  });
+  return (
+    <button
+      data-h2-dialog-trigger={`${id}`}
+      data-h2-button={buttonStyling}
+      ref={ref}
+      type={type || "button"}
+      onClick={onClick}
+      {...props}
+    >
+      {children}
+    </button>
+  );
+};
+
+const Content: React.FunctionComponent<GeneralProps> = (props) => {
+  const { id = "dialog" } = useDialogContext();
+  const { className, children } = props;
+  return (
+    <div
+      data-h2-dialog-content
+      id={`${id}Content`}
+      className={className}
+      {...props}
+    >
+      {children}
+    </div>
+  );
+};
+
+const ExitBtn: React.FunctionComponent<GeneralBtnProps & GeneralProps> = (
+  props,
+) => {
+  useDialogContext(); // Ensures sub-component can only be used within the Dialog component.
+  const { buttonStyling, className, onClick, children } = props;
+  const intl = useIntl();
```

Comment:
The use of react-intl makes this more specific to out Talent Cloud stack, and less useful as an H2 generic component. I still think this component should just use `{children}`, or perhaps a `hiddenText` prop, rather than include specific text.

I also don't really understand how this component is meant to be used? Like what's the difference between this and an ActionBtn, which also causes the dialog to close. It would be helpful if ExitBtn were used in one of the Storybook examples.

Reply:
Ya this is meant for the close button that sits in the top right hand corner of the dialog/modal. You're right it is pretty much the same as the ActionBtn. I'll remove it then, and there will be just one button component within the Dialog.